driver-mapping/src/test/java/com/datastax/driver/mapping/PropertyNamingStrategyTest.java

```diff
@@ -0,0 +1,314 @@
+                .setCassandraConvention(new CommonNamingConventions.LowerCamelCase());
+        MappingManager mappingManager = new MappingManager(session(), conf);
+        Mapper<SnakeCaseToCamelCase> mapper = mappingManager.mapper(SnakeCaseToCamelCase.class);
+        assertThat(mapper.get(1).getMy_xml_parser()).isEqualTo(1);
+        assertThat(mapper.get(1).getOther()).isEqualTo(2);
+    }
+
+    @Table(name = "camel_case_table")
+    public static class SnakeCaseToCamelCase {
+
+        @PartitionKey
+        private int my_xml_parser;
+
+        private int other;
+
+        public int getMy_xml_parser() {
+            return my_xml_parser;
+        }
+
+        public void setMy_xml_parser(int my_xml_parser) {
+            this.my_xml_parser = my_xml_parser;
+        }
+
+        public int getOther() {
+            return other;
+        }
+
+        public void setOther(int other) {
+            this.other = other;
+        }
+
+    }
+
+    @Test(groups = "short")
+    public void test_overrides() {
+        MapperConfiguration conf = new MapperConfiguration();
+        conf.getNamingStrategy()
+                .setJavaConvention(new CommonNamingConventions.LowerSnakeCase())
+                .setCassandraConvention(new CommonNamingConventions.LowerCamelCase());
+        MappingManager mappingManager = new MappingManager(session(), conf);
+        Mapper<TestOverrides> mapper = mappingManager.mapper(TestOverrides.class);
+        assertThat(mapper.get(1).getMyXmlParser()).isEqualTo(1);
+        assertThat(mapper.get(1).getGoo()).isEqualTo(2);
+    }
+
+    @Table(name = "camel_case_table")
+    public static class TestOverrides {
+
+        @PartitionKey
+        @Column(caseSensitive = true)
+        private int myXmlParser;
+
+        @Column(name = "other")
+        private int goo;
+
+        public int getMyXmlParser() {
+            return myXmlParser;
+        }
+
+        public void setMyXmlParser(int myXmlParser) {
+            this.myXmlParser = myXmlParser;
+        }
+
+        public int getGoo() {
+            return goo;
+        }
+
+        public void setGoo(int goo) {
+            this.goo = goo;
+        }
+
+    }@Test(groups = "short")
+
+    public void test_recommended_settings() {
+        MapperConfiguration conf = new MapperConfiguration();
+        conf.getNamingStrategy()
+                .setJavaConvention(new CommonNamingConventions.LowerCamelCase("m", "_"))
+                .setCassandraConvention(new CommonNamingConventions.LowerSnakeCase());
+        MappingManager mappingManager = new MappingManager(session(), conf);
+        Mapper<TestRecommendedSettings> mapper = mappingManager.mapper(TestRecommendedSettings.class);
+        assertThat(mapper.get(1).getmMyXmlParser()).isEqualTo(1);
+        assertThat(mapper.get(1).get_other()).isEqualTo(2);
+    }
+
+    @Table(name = "snake_case_table")
+    public static class TestRecommendedSettings {
+
+        @PartitionKey
+        private int mMyXmlParser;
```

Comment:
So the corresponding table `snake_case_table` contain the fields `my_xml_parser` and `other`. If I am a [sworn android developer](http://source.android.com/source/code-style.html#follow-field-naming-conventions), i'd probably put `m` in the beginning of all my private fields, or i may have any other blind faith [like putting `_`](http://stackoverflow.com/questions/1899683/is-there-a-standard-in-java-for-underscore-in-front-of-variable-or-class-nam) as i personally like. But there many other examples. Anyways, in this test we want to tell the mapper - each time you spot `m` or `_` in the beginning of a field name (or a getter/setter), strip it down. then convert to the output naming conversion.
So what the mapper does here is actually receive `mMyXmlParser` as input, strip it to `myXmlParser`, and then transform to snake case (the output strategy) which results in `my_xml_parser`. Same for `_other` field.

Practically, you should use this option only if you do have a constant prefix you use, and only for the one you use. I don't think anyone actually uses both "_" and "m" as prefixes but the test covers them both. although [some extreme conventions](https://en.wikipedia.org/wiki/Hungarian_notation) have many different prefixes depending on the type of the variable, and we can support it too.